airbyte-webapp/src/components/connectorBuilder/Builder/BuilderFieldWithInputs.tsx

```diff
@@ -0,0 +1,94 @@
+import { faPlus, faUser } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useField } from "formik";
+import { useMemo, useState } from "react";
+
+import { ListBox, ListBoxControlButtonProps, Option } from "components/ui/ListBox";
+
+import { useConnectorBuilderFormState } from "services/connectorBuilder/ConnectorBuilderStateService";
+
+import { getInferredInputs } from "../types";
+import { BuilderField, BuilderFieldProps } from "./BuilderField";
+import styles from "./BuilderFieldWithInputs.module.scss";
+import { InputForm, newInputInEditing } from "./InputsForm";
+
+export const BuilderFieldWithInputs: React.FC<BuilderFieldProps> = (props) => {
+  const [field, , helpers] = useField(props.path);
+
+  return (
+    <BuilderField
+      {...props}
+      adornment={<UserInputHelper setValue={helpers.setValue} currentValue={field.value} />}
+      className={styles.inputWithHelper}
+    />
+  );
+};
+
+export const UserInputHelper = ({
+  setValue,
+  currentValue,
+}: {
+  setValue: (value: string) => void;
+  currentValue: string;
+}) => {
```

Comment:
I just forgot to add it here, but it's more a code-style thing in this scenario.

Using the React.FC type is more explicit about this being a function component which improves typing, but only a very little bit.

For example take a look at this "function":
```
const MyFn = () => {
  if (Math.random() < 0.5) return undefined;
  return <span></span>;
};
```

it's actually slightly invalid because a function component is not allowed to return undefined (has to be null). However, as this is just a function as far as Typescript is concerned, there won't be a type error at definition time (it will complain if you actually try to use it though).

If you do 
```
const MyFn: React.FC = () => {
  if (Math.random() < 0.5) return undefined;
  return <span></span>;
};
```

however, it will complain directly with
```
Type '() => JSX.Element | undefined' is not assignable to type 'FC<{}>'.
  Type 'Element | undefined' is not assignable to type 'ReactElement<any, any> | null'.
    Type 'undefined' is not assignable to type 'ReactElement<any, any> | null'.ts(2322)
```

Also, there are a few additional keys you can set on the function object itself (functions are also objects in Javascript, so you can set arbitrary stuff on them) that are picked up by react:
```
    interface FunctionComponent<P = {}> {
        (props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | null;
        propTypes?: WeakValidationMap<P> | undefined;
        contextTypes?: ValidationMap<any> | undefined;
        defaultProps?: Partial<P> | undefined;
        displayName?: string | undefined;
    }
```

However, they are not really important in our setup as it's mostly about an ad-hoc runtime typing React can do (which we don't really need as all our code is typed via Typescript already).

Reply:
Got it, that makes sense. So it is probably better to always defer to using `React.FC` since it applies stricter typing constraints and will therefore fail faster